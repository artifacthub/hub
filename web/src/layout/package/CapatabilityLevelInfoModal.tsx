import React, { useState } from 'react';
import { MdInfoOutline } from 'react-icons/md';

import Loading from '../common/Loading';
import Modal from '../common/Modal';
import styles from './CapatabilityLevelInfoModal.module.css';

const CapatabilityLevelInfoModal = () => {
  const [onLoadedImage, setOnLoadedImage] = useState<boolean>(false);

  return (
    <Modal
      className="d-none d-lg-inline-block mt-1 ml-2"
      buttonContent={<MdInfoOutline />}
      buttonType={`btn-link btn-sm px-0 pb-0 text-secondary ${styles.btn}`}
      header={<div className="h5 m-0 flex-grow-1">Capability level</div>}
    >
      <div className="my-3 mw-100">
        <div className={styles.imageWrapper}>
          {!onLoadedImage && <Loading />}
          <img
            src="/static/media/capability-level-diagram.svg"
            alt="Capability Level Diagram"
            className="capability-level-diagram mw-100"
            onLoad={() => setOnLoadedImage(true)}
          />
        </div>
      </div>
    </Modal>
  );
};

export default CapatabilityLevelInfoModal;
