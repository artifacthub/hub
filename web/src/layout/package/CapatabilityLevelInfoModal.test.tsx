import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import CapatabilityLevelInfoModal from './CapatabilityLevelInfoModal';

describe('CapatabilityLevelInfoModal', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const result = render(<CapatabilityLevelInfoModal />);
    expect(result.asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders component', () => {
      const { getByText } = render(<CapatabilityLevelInfoModal />);
      expect(getByText('Capability level')).toBeInTheDocument();
    });

    it('opens modal', () => {
      const { getByRole, getByTestId, getByAltText } = render(<CapatabilityLevelInfoModal />);

      expect(getByRole('dialog')).not.toHaveClass('active');
      const btn = getByTestId('openModalBtn');
      fireEvent.click(btn);

      expect(getByRole('dialog')).toHaveClass('active');
      expect(getByAltText('Capability Level Diagram')).toBeInTheDocument();
    });
  });
});
