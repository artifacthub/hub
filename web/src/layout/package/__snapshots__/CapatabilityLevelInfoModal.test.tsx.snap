// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CapatabilityLevelInfoModal creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-none d-lg-inline-block mt-1 ml-2"
  >
    <div
      class="position-relative buttonWrapper"
    >
      <button
        class="font-weight-bold text-uppercase position-relative btn btn-block btn btn-link btn-sm px-0 pb-0 text-secondary btn"
        data-testid="openModalBtn"
        type="button"
      >
        <div
          class="d-flex align-items-center justify-content-center"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="modal modal"
      role="dialog"
    >
      <div
        class="modal-dialog modal-lg modal-dialog-centered modal-dialog-scrollable"
      >
        <div
          class="modal-content content"
        >
          <div
            class="modal-header d-flex flex-row align-items-center header undefined"
          >
            <div
              class="h5 m-0 flex-grow-1"
            >
              Capability level
            </div>
            <button
              class="close"
              data-testid="closeModalBtn"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
          <div
            class="modal-body p-4 h-100 d-flex flex-column"
          >
            <div>
              <div
                class="alertWrapper"
                data-testid="alertWrapper"
              />
            </div>
          </div>
          <div
            class="modal-footer p-3"
          >
            <button
              class="btn btn-sm btn-secondary text-uppercase"
              data-testid="closeModalFooterBtn"
              type="button"
            >
              <div
                class="d-flex flex-row align-items-center"
              >
                <svg
                  class="mr-2"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <div>
                  Close
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
